pkgs/build-support/writers/default.nix

```
@@ -94,8 +94,8 @@ rec {
         ]}
         gcc \
             ${optionalString (libraries != [])
-              "$(pkgs.pkgconfig}/bin/pkg-config --cflags --libs ${
-                concatMapStringsSep " " (lib: escapeShellArg (builtins.parseDrvName lib.name).name) (libraries)
+              "$(pkg-config --cflags --libs ${
+                concatMapStringsSep " " (pkg: "$(find ${escapeShellArg pkg}/lib/pkgsconfig -name \*.pc -exec basename {} \;)") libraries
```

gnufind also supports the following:

```suggestion
                concatMapStringsSep " " (pkg: "$(find ${escapeShellArg pkg}/lib/pkgsconfig -name \*.pc -printf "%f")") libraries
```

---

ah cool!
since this is a backport I will open a new PR with the suggested changes?

---

Yeah that sounds good, make a PR for the suggestions for master and add the commit to this PR as a cherry-pick as well